component/pyroscope/scrape/target.go

```diff
@@ -400,3 +400,28 @@ func targetsFromGroup(group *targetgroup.Group, cfg Arguments) ([]*Target, []*Ta
 
 	return targets, droppedTargets, nil
 }
+
+var k8sNameLabels = []string{
+	"__meta_kubernetes_pod_label_pyroscope_io_name",
```

Comment:
The way I see it: a diligent user is supposed to create a relabeling config to populate `service_name` from an annotation/label or put it somehow else. If we provide a way to avoid this extra config by putting `pyroscope.io/name` label (for consistency, may be we should call it `pyroscope.io/service-name`), it would be nice to have it mentioned in the documentation.

Reply:
lets go with the concatenation (and risk the repetitions)

Reply:
Yeah, that works. But I wouldn't defer to `pod_container_name` and `pod_name` labels even as a last resort. Moreover, I'd question necessity of falling back to "unknown" service. Maybe we should give up early.

---

On a side note: k8s labels vs annotations – I think it may be slightly better to use annotations only for our purposes. I understand that not everyone follows the rules, but still.

[Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/): 
> You can use Kubernetes annotations to attach arbitrary non-identifying metadata to [objects](https://kubernetes.io/docs/concepts/overview/working-with-objects/#kubernetes-objects). Clients such as tools and libraries can retrieve this metadata.

[Labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/):
> Labels are intended to be used to specify identifying attributes of objects that are meaningful and relevant to users, but do not directly imply semantics to the core system.

Reply:
if "give up early" what happens from a user perspective?

Reply:
1. No profiling data appears in the UI. So the user don't need to clean up after the unsuccessful attempt
2. A message indicating the misconfiguration is written to the agent log every time the target is discovered

Reply:
I like the idea to give up early and not being clever for scrape.

For ebpf I still think we should go with concatenation or something else

Reply:
RE: eBPF concatenation, I don't think we need `namespace`, since it already be there (if you can infer via PID -> service discovery), right?

And based on the same principle, you can also look up the same tags you are looking for normal scrape (service_name, name etc).

Then as a last resort, the binary name could be used (somebody suggested it before), no?